core/tests/unittests/utils/test_utils.py

```diff
@@ -0,0 +1,68 @@
+import unittest
+
+import numpy as np
+import pandas as pd
+
+from autogluon.core.utils import infer_problem_type
+from autogluon.core.constants import BINARY, MULTICLASS, MULTICLASS_UPPER_LIMIT, REGRESSION
+
+
+class TestInferProblemType(unittest.TestCase):
+
+    def test_infer_problem_type_empty(self):
+        with self.assertRaises(ValueError):
+            infer_problem_type(pd.Series([], dtype="float"))
+
+    def test_infer_problem_type_non_finite(self):
+        with self.assertRaises(ValueError):
+            infer_problem_type(pd.Series([np.nan, np.inf, np.NINF]))
```

Comment:
Can we split this one into 3 different cases (since each one is special)?

Reply:
Updated in new revision